code/datums/components/plumbing/_plumbing.dm

```diff
@@ -93,30 +93,23 @@
 
 ///called from in process(). only calls process_request(), but can be overwritten for children with special behaviour
 /datum/component/plumbing/proc/send_request(dir)
-	process_request(dir = dir)
+	process_request(amount = MACHINE_REAGENT_TRANSFER, reagent = null, dir = dir)
 
 ///check who can give us what we want, and how many each of them will give us
-/datum/component/plumbing/proc/process_request(amount = MACHINE_REAGENT_TRANSFER, reagent, dir)
-	//find the duct to take from
+/datum/component/plumbing/proc/process_request(amount, reagent, dir)
+	var/list/valid_suppliers = list()
 	var/datum/ductnet/net
 	if(!ducts.Find(num2text(dir)))
 		return
 	net = ducts[num2text(dir)]
-
-	//find all valid suppliers in the duct
-	var/list/valid_suppliers = list()
 	for(var/datum/component/plumbing/supplier as anything in net.suppliers)
 		if(supplier.can_give(amount, reagent, net))
 			valid_suppliers += supplier
+	// Need to ask for each in turn very carefully, making sure we get the total volume. This is to avoid a division that would always round down and become 0
+	var/targetVolume = reagents.total_volume + amount
```

Comment:
Sorry to be that guy but I'd rather this left in snake case. There should just not be camel case in dm